### DISCUSSION TITLE
Start Zim in read-only mode

### DISCUSSION THREAD

**Discussion by Spacewalker2** [zim-desktop-wiki/zim-desktop-wiki#2095] [Q&A] [answered]
Hello,

I encountered the "Toggle Editable" button where I can switch to read-only. For some notebooks I like this very much. Is there a way to start Zim with read-only enabled, so I have to toggle first before I make some changes?

By the way thanks for this great tool!

**Comment by jaap-karssenberg** [comment]
The state of the editable button is remembered when you close the notebook;
so if you turn it off after editing it is still off next time you start up.

However, I can imagine a "default read-only" flag per notebook that ensures
it is always off when opening the notebook. Maybe even with this option
turned on, we should re-set the read-only mode after editing a page and
moving to a next page.

--- Jaap

**Comment by Spacewalker2** [comment]
> The state of the editable button is remembered when you close the notebook;
> so if you turn it off after editing it is still off next time you start up.

Unfortunately this is not the case for me. When I turn read-only mode
on the next time I open the notebook read-only is off again. The machine
where I tried this is a Manjaro Linux with XFCE desktop environment. Maybe
there is some missing package or functionality.

> However, I can imagine a "default read-only" flag per notebook that ensures
> it is always off when opening the notebook. Maybe even with this option
> turned on, we should re-set the read-only mode after editing a page and
> moving to a next page.

This sounds like a great idea, but you mean it ensures that it is always
in read-only mode right?

**Comment by jaap-karssenberg** [comment, marked as answer]
The state saving is fixed in development branch, will work in next release
:)

My idea would be to still allow editing a notebook by using the "editable"
toggle. However, than also switch back that toggle automatically e.g. when
you leave the page, or did not edit for x minutes, or ...  (Actually, with
that "switch back" logic in place, the startup is not special anymore - it
just relies on "switch back" + save state)

--- Jaap

**Reply by Spacewalker2** [reply]
> The state saving is fixed in development branch, will work in next release
> :)

These are good news. Thanks!

> My idea would be to still allow editing a notebook by using the "editable"
> toggle. However, than also switch back that toggle automatically e.g. when
> you leave the page, or did not edit for x minutes, or ...  (Actually, with
> that "switch back" logic in place, the startup is not special anymore - it
> just relies on "switch back" + save state)

This sounds great. It would be nice to have that.
